freqtrade/optimize/backtesting.py

```diff
@@ -275,16 +275,20 @@ def _get_ohlcv_as_lists(self, processed: Dict[str, DataFrame]) -> Dict[str, Tupl
             # Trim startup period from analyzed dataframe
             df_analyzed = processed[pair] = pair_data = trim_dataframe(
                 df_analyzed, self.timerange, startup_candles=self.required_startup)
+            # Update dataprovider cache
+            self.dataprovider._set_cached_df(pair, self.timeframe, df_analyzed)
+
+            # Create a copy of the dataframe before shifting, that way the buy signal/tag
+            # remains on the correct candle for callbacks.
+            df_analyzed = df_analyzed.copy()
```

Comment:
@rokups you think this (combined with the comment above) is clear enough, or should we use a different variable name to make it clear that we're using a copy from here onwards?

Reply:
Looks clear enough to me. Im not sure we will ever be in a situation asking ourselves why our changes do not propagate to dataframe passed to strategies. Because there should be no such changes

Reply:
no, i'm more worried about removal of this "copy" as it would appear unnecessary (maybe as part of refactorings or whatnot).

Reply:
Comment states the reason why copy exists. Why would anyone think it is unnecessary when comment says otherwise?

Reply:
because i don't always read comments :rofl: